simulation/halsim_gui/src/main/native/cpp/Assembly2D.cpp

```diff
@@ -0,0 +1,249 @@
+#include <units/units.h>
+#include <wpi/Path.h>
+#include <wpi/SmallString.h>
+#include <wpi/json.h>
+#include <wpi/raw_istream.h>
+#include <wpi/raw_ostream.h>
+
+
+#include "GuiUtil.h"
+#include "HALSimGui.h"
+#include "SimDeviceGui.h"
+#include "portable-file-dialogs.h"
+#include <mockdata/EncoderData.h>
+
+#include <vector>
+#include <map>
+#include <list>
+#include <list>
+#include <string>
+#include <cmath>
+
+using namespace halsimgui;
+
+int windowWidth = 100;
+int windowHeight = 100;
+
+static std::vector<std::string> listOfColors = {"white", "silver", "gray", "black", "red", "maroon", "yellow", "olive",
+                                                "lime",
+                                                "green", "aqua", "teal", "blue", "navy", "fuchsia", "purple"};
+
+static std::vector<ImColor> listOfRGB = {IM_COL32(255, 255, 255, 255), IM_COL32(192, 192, 192, 255),
+                                         IM_COL32(128, 128, 128, 255), IM_COL32(0, 0, 0, 255), IM_COL32(255, 0, 0, 255),
+                                         IM_COL32(128, 0, 0, 255), IM_COL32(255, 255, 0, 255),
+                                         IM_COL32(128, 128, 0, 255), IM_COL32(0, 255, 0, 255), IM_COL32(0, 128, 0, 255),
+                                         IM_COL32(0, 255, 255, 255),
+                                         IM_COL32(0, 128, 128, 255), IM_COL32(0, 0, 255, 255), IM_COL32(0, 0, 128, 255),
+                                         IM_COL32(255, 0, 255, 255), IM_COL32(128, 0, 128, 255)};
+
+std::map<std::string, ImColor> colorLookUpTable;
+
+
+static void buildColorTable() {
+    for (int i = 0; i < (int) listOfColors.size(); i++) {
+        colorLookUpTable[listOfColors[i]] = listOfRGB[i];
+    }
+}
+
+struct BodyConfig {
+    std::string name;
+    std::string type = "line";
+    int startLocation = 0;
+    int length = 100;
+    std::string color = "green";
+    int angle = 0;
+    std::list<BodyConfig> children;
+    int lineWidth = 1;
+};
+
+std::list<BodyConfig> bodyConfigList;
+
+std::tuple<float, float, float>
+DrawLine(int startXLocation, int startYLocation, int length, double angle, ImDrawList *drawList, ImVec2 windowPos,
+         ImColor color, const BodyConfig &bodyConfig, const std::string &previousPath) {
+    // Find the current path do the ligament
+    std::string currentPath = previousPath + bodyConfig.name;
+    // Find the angle in radians
+    double radAngle = (angle - 90) * 3.14159 / 180;
+    // Get the start X and Y location
+    double xEnd = startXLocation + length * std::cos(radAngle);
+    double yEnd = startYLocation + length * std::sin(radAngle);
+    // Add the line to the drawList
+    drawList->AddLine(windowPos + ImVec2(startXLocation, startYLocation),
+                      windowPos + ImVec2(xEnd, yEnd),
+                      color, bodyConfig.lineWidth);
+    // Return the end X, Y, and angle
+    return {xEnd, yEnd, angle};
+}
+
+HAL_SimDeviceHandle m_devHandle = 0;
```

Comment:
This should be static.  Not sure why the m_ prefix?  Did you want to put all of this into a class?